### PR TITLE
Add MachineAddress type for infrastructure providers

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR adds a new type copied from `corev1.NodeAddress`: `MachineAddress`.
Infrastructure Providers should use this type in their Status to store the Machine's addresses.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/apis/cluster/v1alpha2/common_types.go [hunk 1]
```diff
@@ -20,6 +20,29 @@ import (
...
 
+// MachineAddressType describes a valid MachineAddress type.
+type MachineAddressType string
+
+const (
+	NodeHostName    MachineAddressType = "Hostname"
+	NodeExternalIP  MachineAddressType = "ExternalIP"
+	NodeInternalIP  MachineAddressType = "InternalIP"
+	NodeExternalDNS MachineAddressType = "ExternalDNS"
+	NodeInternalDNS MachineAddressType = "InternalDNS"
```
Comment (detiber) [inline pkg/apis/cluster/v1alpha2/common_types.go]: s/Node/Machine/ ?

Comment (vincepri) [reply pkg/apis/cluster/v1alpha2/common_types.go]: Fixed

**Pull request merged**